pkg/sql/execinfrapb/processors_bulk_io.proto

```diff
@@ -255,22 +255,33 @@ message SplitAndScatterSpec {
   repeated TableRekey rekeys = 2 [(gogoproto.nullable) = false];
 }
 
-// FileCompression list of the compression codecs which are currently
-// supported for CSVWriter spec
+// FileCompression is a list of the compression codecs which are currently
+// supported for Exporter spec.
 enum FileCompression {
   None = 0;
   Gzip = 1;
+  Snappy = 2; // Only available for Parquet.
 }
 
-// CSVWriterSpec is the specification for a processor that consumes rows and
-// writes them to CSV files at uri. It outputs a row per file written with
+// FileFormat is list of file formats which are currently
+// supported for the Exporter spec.
+/*enum FileFormat {
+  Csv = 0;
+  Parquet = 1;
+}*/
+
+// ExporterSpec is the specification for a processor that consumes rows and
+// writes them to Parquet or CSV files at uri. It outputs a row per file written with
 // the file name, row count and byte size.
 message CSVWriterSpec {
   // destination as a cloud.ExternalStorage URI pointing to an export store
   // location (directory).
   optional string destination = 1 [(gogoproto.nullable) = false];
   optional string name_pattern = 2 [(gogoproto.nullable) = false];
-  optional roachpb.CSVOptions options = 3 [(gogoproto.nullable) = false];
+  //optional FileFormat file_format = 10 [(gogoproto.nullable) = false];
+  optional roachpb.CSVOptions csv_options = 3 [(gogoproto.nullable) = true];
+  optional roachpb.ParquetOptions parquet_options = 8 [(gogoproto.nullable) = true];
```

Comment:
Now that the Parquet Processor and the CSV Processor use the same kind of spec, I'm wondering if there's any value passing a FileFormat Enum to the spec (a few lines above); or should I stick to my current implementation which maintains the following in the spec: CSVOptions will be not nil IFF the spec is getting passed the CSV Exporter. This same invariant applies to ParquetOptions.

Reply:
Should I worry about modifying CSVOptions, a current field in this proto message. Do I need to bump to the processor version in version.go?

Reply:
> I'm wondering if there's any value passing a FileFormat Enum to the spec (a few lines above); or should I stick to my current implementation which maintains the following in the spec: CSVOptions will be not nil IFF the spec is getting passed the CSV Exporter. This same invariant applies to ParquetOptions.

The current implementation seems OK to me. There is some prior art in the existing protobufs that make use of either protobuf's `oneof` or gogoproto.onlyone that might be good to take a peak at if you are looking for alternatives -- although I don't know our typical style here.

Reply:
discovered roachpb.IO_Formats. Just pushed a significant refactor. Conversation no longer applies.